enhancements/hypershift/hostedClusterInfrastructure.md

```diff
@@ -0,0 +1,139 @@
+  1. Infrastructure can be pre-created using an ACM Cloud Provider credential or by pre-populating the Custom Resource with values and applying it to the ACM/MCE hub. These can then be consumed by the ACM/MCE UI or CLI, and is a nice option, as the AWS credential does not need to be exposed to the user creating the HostedCluster.
+  2. You can re-use the HostedClusterInfrastructure resource between HostedClusters. This is especially useful in Azure, where it takes longer to create infrastructure then the hosted control plane.
+
+### User Stories
+- As a Cluster Administrator I want to create one or more HostedClusterInfrastructure resources for my team to consume
+- As a Cluster Administrator, I want to choose an ACM Cloud Provider credential and deploy a Cloud Provider HostedCluster from start to finish
+- As a Cluster Consumer, I want to choose an available HostedClusterInfrastructure resource and have a HostedCluster deploy, without knowing or having to provide an AWS credential
+
+### Goals
+- Provide a consumable API for HostedClusterInfrastructure that satisfies existing ACM/MCE use cases and expected user stories(flows)
+
+### Non-Goals
+- The consumer usecase also compliments the cluster create templating flow
+
+## Proposal
+### API
+[HostedClusterInfrastructure](https://github.com/openshift/hypershift/blob/27c0a432bdc8d702f1cdb2a2f3f25e5ae6fbee7d/api/v1alpha1/hostedclusterinfrastructure_types.go) is the consumer facing API exposed for Cloud Infrastructure creation and destroy.
+Its reason to exist is to preserve Red Hat Advanced Cluster Management user stories and allows our ability to satisfy consumer needs and evolve at our own pace based on user feedback.
+
+#### Implementation
+The HostedClusterInfrastructure controller is meant to be limited interface, that leverages the same CLI flows (so they are always compatable) to create, destroy and evaluate (spike) Cloud Provider infrastructure resources when the Cloud Provider credential is available.
+
+When a new HostedClusterInfrastructure is created:
+- The HostedClusterInfrastructure controller will reconcile
+  - First it will determine if a Cloud Provider credential is present
+    - If Present:
+      - It will create the infrastructure where needed and validate the infrastructure where it exists
+    - If Not Present
+      - The resource just contains user provided values that can be consumed by the UI or CLI
+- CLI will be able to create or render the HostedClusterInfrastructure resource
+- CLI will be able to consume the HostedClusterInfrastructure resource
+- User may create/apply the resource with all values
+- Delete HostedClusterInfrastructure the controller will reconcile
+  - First it will determine if a Cloud Provider credential is present
+    - If Present:
+      - It will delete all the infrastructure resources found in the cr before allowing the cr to be removed
+    - If Not Present
+      - The cr will just be deleted
```

Comment:
in a situation where the credential existed at one point, but was deleted before the HCI, wouldn't this kind of "silently" leave infra orphaned? If so, should we account for this?

Reply:
Yes, in this scenario, there is a status check for the cloud provider (borrowing how it looked in HypershiftDeployment), and in the status for the HostedClusterInfrastructure, we want to differentiate whether the resource was applied successfully or just provided without ever being applied.  We would detect this case an show an appropriate error and have a finalizer blocking deletion of the HostedClusterResource.